tests/functional/aws-node-sdk/test/object/post.js

```diff
@@ -0,0 +1,1621 @@
+                    done();
+                } catch (err) {
+                    done(err);
+                }
+            });
+
+            req.on('error', err => {
+                done(err);
+            });
+
+            formData.pipe(req);
+            return undefined;
+        });
+    });
+
+    it('should handle url invalid characters in keys', done => {
+        const { url } = testContext;
+        const fields = calculateFields(ak, sk, [{ key: 'key with spaces' }]);
+        const formData = new FormData();
+        const encodedKey = 'key%20with%20spaces'; // Expected URL-encoded key
+
+        fields.forEach(field => {
+            formData.append(field.name, field.value);
+        });
+
+        formData.append('file', fileBuffer, { filename });
+
+        formData.getLength((err, length) => {
+            if (err) {
+                return done(err);
+            }
+
+            const options = {
+                method: 'POST',
+                hostname: url.hostname,
+                port: url.port,
+                path: url.pathname + url.search,
+                headers: {
+                    ...formData.getHeaders(),
+                    'Content-Length': length,
+                },
+            };
+
+            const req = http.request(options);
+
+            req.on('response', res => {
+                try {
+                    assert.equal(res.statusCode, 204);
+                    assert.equal(res.headers.location, `${url.hostname}:${url.port}/${bucketName}/${encodedKey}`);
+                    done();
+                } catch (err) {
+                    done(err);
+                }
+            });
+
+            req.on('error', err => {
+                done(err);
+            });
+
+            formData.pipe(req);
+            return undefined;
+        });
+    });
+
+    it('should handle error when bucket does not exist', done => {
+        const fakeBucketName = generateBucketName();
+        const tempUrl = `${config.endpoint}/${fakeBucketName}`;
+        const fields = calculateFields(ak, sk, [], fakeBucketName);
+        const formData = new FormData();
+
+        fields.forEach(field => {
+            formData.append(field.name, field.value);
+        });
+
+        formData.append('file', fileBuffer, { filename });
+
+        formData.getLength((err, length) => {
+            if (err) {
+                return done(err);
+            }
+
+            const parsedUrl = new URL(tempUrl);
+
+            const options = {
+                method: 'POST',
+                hostname: parsedUrl.hostname,
+                port: parsedUrl.port,
+                path: parsedUrl.pathname + parsedUrl.search,
+                headers: {
+                    ...formData.getHeaders(),
+                    'Content-Length': length,
+                },
+            };
+
+            const req = http.request(options);
+
+            req.on('response', res => {
+                if (res.statusCode === 404) {
```

Comment:
ideally it should look like `assert.strictEqual(res.statusCode, 404)` so it's easy to see in the logs tha actual value vs expected